lib/sycamore/sycamore/query/operators/query_database.py

```diff
@@ -42,3 +42,54 @@ class QueryDatabase(LogicalOperator):
     Whenever possible, use the query parameter to filter data at the source, as this is more
     efficient than filtering data in subsequent data filtering operators.
     """
+
+
+class QueryVectorDatabase(LogicalOperator):
+    """OpenSearch knn query to load records that are similar to a given query string. Returns the top k records
+    according to the cosine similarity between the query string and record's text content.
+    Use this if we need to filter on a field that can't suffice with an exact match, but also isn't complex enough to
+    require a llm_filter, i.e. a record level LLM call.
+
+    Here are some instructions about using QueryVectorDatabase instead of QueryDatabase:
+    1. Use QueryVectorDatabase if the query plan uses an LLM operator (LLMFilter or LLMExtractEntity),
+        and the final answer doesn't require you to scan the full dataset (e.g. find me an instance of a transcript
+        that talked about AI)
+    2. Use QueryVectorDatabase for RAG-style questions where you need to retrieve a set of candidate records where the
+        answer might exist.
+    3. Do not use QueryVectorDatabase where examining each record is essential (e.g. did all transcripts talk about AI?)
+    4. Do not use QueryVectorDatabase where you need to potentially return large datasets (e.g. give me all transcripts
+        that talked about AI)
+
+    """
+
+    index: str
+    """The index to load data from."""
+
+    query_to_embed: str
+    """The string to convert to a vector and perform vector search"""
+
+    filter: Optional[Dict] = None
+    """
+    A filter query in OpenSearch Query DSL format which represents a filter in a knn query. 
```

Comment:
Same as above, I think the wording here should probably not mention "knn query"

Reply:
I updated this param to say 'opensearch_filter', and left opensearch specific wording in here